packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -81,24 +81,72 @@ import ResetPasswordContainer from '../../pages/ResetPasswordRedesign/ResetPassw
 import ConfirmResetPasswordContainer from '../../pages/ResetPasswordRedesign/ConfirmResetPassword/container';
 import CompleteResetPasswordWithCodeContainer from '../../pages/ResetPasswordRedesign/CompleteResetPassword/container';
 import AccountRecoveryConfirmKeyContainer from '../../pages/ResetPasswordRedesign/AccountRecoveryConfirmKey/container';
+import SignoutSync from '../../pages/SignoutSync';
 
 const Settings = lazy(() => import('../Settings'));
 
 export const App = ({
   flowQueryParams,
 }: { flowQueryParams: QueryParams } & RouteComponentProps) => {
   const config = useConfig();
+  const session = useSession();
+  const integration = useIntegration();
+  const isSync = integration != null && integration.isSync();
+  const { data: isSignedInData } = useLocalSignedInQueryState();
 
   // GQL call for minimal metrics data
   const { loading: metricsLoading, data } = useInitialMetricsQueryState() ?? {};
 
-  // Because this query depends on the result of an initial query (in this case,
-  // metrics), we need to run it separately.
-  const { data: isSignedInData } = useLocalSignedInQueryState();
-  const integration = useIntegration();
+  // Determine if user is actually signed in
+  const [isSignedIn, setIsSignedIn] = useState<boolean | undefined>(undefined);
+  useEffect(() => {
+    if (!integration) {
+      return;
+    }
+
+    // If the local appollo cache says we are signed in, then we can skip the rest.
+    if (isSignedInData?.isSignedIn === true) {
+      setIsSignedIn(true);
+      return;
+    }
+
+    (async () => {
+      let isValidSession = false;
+
+      if (isSync) {
+        const syncUser = await firefox.requestSignedInUser(
+          integration!.data.context
+        );
+
+        if (syncUser) {
+          // If the session is valid, try to set it as the current account
+          isValidSession = await session.isValid(syncUser.sessionToken);
```

Comment:
Should we be re-running the initial metrics query with this session token and if so, could we rely on the response from that query to determine if it's valid, similar to what we're doing now?

Reply:
Let me double check this... When I was manually testing it seemed like session.isValid, will trigger a rerender when it writes to the cache, and metrics would then be correct. 

That being said, maybe the initial metrics query shouldn't run until this happens? It might just make the code more 'obvious'. I'm happy to take another pass on this, cause it was something that came to mind.

Reply:
In addition to changing the appollo cache. We setIsSignedIn on line 133 which will also trigger the rerender.